Blog: The most useful customer retention metrics, ranked (#8391)
PostHog/posthog.com: contents/product-engineers/best-customer-retention-metrics.md

For example, if your ARPU is $50 per month and your average customer lifespan is 12 months, then your CLTV would be $50 x 12, or $600.

### What’s a good CLTV?

To work this out you need to know your CAC. That’s your customer acquisition cost, or, simply put, the average cost to your business of snagging a new paying customer. If your CAC is lower than your CLTV then you’ve got a healthy business that’s able to acquire new paying customers without losing money. If it’s not, you're in trouble.

As a guide, at early-stage startups the CLTV might be lower as you acquire new customers and find product-market fit. According to data compiled by search optimization firm [First Page Sage](https://firstpagesage.com/seo-blog/the-ltv-to-cac-ratio-benchmark/), the average CLTC to CAC ratio at B2B SaaS companies is 4:1 At B2C SaaS companies it’s 2.5:1.

And this kind of healthiness compounds: acquiring customers without spending a ton on marketing allows you to focus more on improving and expanding your existing product, delighting your most loyal customers, and improving overall customer satisfaction.

## 3. Revenue churn rate

### What is revenue churn rate?

> It’s not just customers who churn, revenue churns with them. This metric helps you understand how much money your business makes from each user over time. It can help you spot where you’re losing revenue due to downgrades, people failing to renew, or, worst of all, cancellations. 
> 
> While your customer churn rate looks at the number of customers you lose in a given period, your revenue churn specifically focuses on the dollar value of recurring period lost, or gained, in a certain period.
> 
> ### How is it useful?
> 
> Any subscription-based business should be measuring their revenue churn rate. These days, that’s pretty much all businesses from coffee subscription businesses to B2B SaaS companies and everything in between. Businesses that sell big, high-value items (think houses, industrial equipment) likely don’t need to focus on revenue churn.

Reviewer: ```suggestion
It’s not just customers who churn, revenue churns with them. 

While your customer churn rate looks at the *number* of customers you lose in a given period, revenue churn specifically focuses on the *dollar value* of recurring revenue lost, or gained, in a certain period.

### Why is it useful?

Because it helps you understand how much money your business makes from each user over time, and the impact of customer churn on revenue.

Not all revenue churn comes from customers failing to renew, or cancelling their contract. It can also come from customers *downgrading* their subscription, or using your product less.

This is really useful for understanding the exact nature of retention and churn in your product. A company, for example, could have a low customer churn rate, but high revenue churn. This would suggest either:

1. Only high-value customers are churning.
2. Customers are choosing to spend less / downgrade, rather than churn entirely.

Thus, analyzing customer retention and revenue churn together is essential to finding the truth about your business.
```

### How to calculate revenue churn rate

> You can calculate your revenue churn rate based on either gross or net revenue. A focus on gross revenue is simpler as it’ll show you how much revenue is churning out of your business, but a focus on net revenue will show both the amount of revenue lost from, say, customers downgrading, but also the revenue gained from customer upgrading.

Reviewer: ```suggestion
You can calculate your revenue churn rate based on either gross or net revenue: 

- **Gross revenue** is simpler but less useful, as it’ll show you how much revenue is churning out of your business without much context. 

- **Net revenue** is harder to calculate, but way more powerful because it'll show both the amount of revenue lost from, say, customers downgrading, but also the revenue gained from customers upgrading.
```

> Let’s do the math. For your gross revenue churn rate, take your churned revenue in a month and divide it by the monthly recurring revenue (MRR) from the previous month, and multiply that total by 100. 

Reviewer: ```suggestion
To calculate gross revenue churn rate, take your churned revenue in a month and divide it by the monthly recurring revenue (MRR) from the previous month, and multiply that total by 100. 
```

> For example, Hogflix, an imaginary hedgehog-themed Netflix competitor, had an MRR of $100,000 in May. In April, its churned revenue was $17,000. Divide $17,000 by $100,000 to get 0.17, then multiply that by 100. Voila, Hogflix has a gross revenue churn rate of 17%.

Reviewer: ```suggestion
For example, Hogflix, a streaming platform for hedgehogs, had an MRR of $100,000 in May. In April, its churned revenue was $17,000. Divide $17,000 by $100,000 to get 0.17, then multiply that by 100. Voila, Hogflix has a gross revenue churn rate of 17%.
```

> To work out your net revenue churn, find your churned revenue in a period and subtract your expansion revenue and divide that total by your monthly recurring revenue in the previous period. Now multiply that new total by 100.

Reviewer: ```suggestion
To calculate net revenue churn, find your churned revenue in a period and subtract your expansion revenue and divide that total by your monthly recurring revenue in the previous period. Now multiply that new total by 100.
```

> For example, Hogflix had an MRR of $100,000 in May. In April, its churned revenue was $17,000 and its expansion revenue was $3,000. So subtract $3,000 from $17,000 to get $15,000 and divide that by $100,000 to get 0.15. Multiply that by 100 to get 15%.

Reviewer: ```suggestion
For example, Hogflix had MRR of $100,000 in May. In April, its churned revenue was $17,000 and its expansion revenue was $3,000. So, subtract $3,000 from $17,000 to get $15,000 and divide that by $100,000 to get 0.15. Multiply that by 100 to get 15%.
```

Reviewer: I think we'll need an illustration here to make this clearer

### What’s a good revenue churn rate?

This depends again on how much your customers pay and how often they pay. If your business is based on monthly subscriptions, and your average revenue per account is $10 or less per month, then aim for monthly net revenue churn of around 6% or less. 

> The more customers pay, the lower your churn should be. At $250 average revenue per account per month, aim for a net revenue churn [of 1.4% or less](https://techcrunch.com/2022/04/20/study-up-on-churn-rate-basics-to-set-customer-and-revenue-benchmarks/), according to data compiled by analytics platform ChartMogul. This sort of churn rate also assumes that your customers are paying annually and, therefore, churn less often because they can’t ditch their contract.

Reviewer: I think this whole section is missing some important context. Net revenue retention can be negative, right? And that would be a very good thing for a SaaS business, which is our main audience, so we should talk about that a bit more.

As an additional, I wonder if we should have something about net / gross revenue retention. Afaik, they're basically the same thing but express slightly differently. I know we use NRR specifically. Good bit on it here: https://www.lightercapital.com/blog/net-revenue-retention-vs-gross-revenue-retention

## 4. Net Promoter Score (NPS)

### What is NPS? 

Developed in the early-2000s and popularized by an article in _[Harvard Business Review](https://hbr.org/2003/12/the-one-number-you-need-to-grow)_, the Net Promoter Score continues to be a very popular way of measuring customer loyalty. It’s based on one question: ‘How likely are you to recommend [company/product/service] to a friend or colleague on a scale of 0 to 10?’Customers are then put into one of three buckets. 

> * **Promoters**, who gave a score of 9 or 10, are loyal enthusiasts and very likely to recommend you. 
> * **Passives**, who gave a score of 7 or 8, aren’t super happy or unhappy but are also unlikely to promote your business. 
> * **Detractors**, those who rated you between 0 and 6, could potentially damage your business through word-of-mouth.

Reviewer: ```suggestion
* **Promoters**, who gave a score of 9 or 10, are loyal enthusiasts and very likely to recommend you.
 
* **Passives**, who gave a score of 7 or 8, aren’t super happy or unhappy but are also unlikely to promote your business.
 
* **Detractors**, those who rated you between 0 and 6, could potentially damage your business through word-of-mouth.
```

As you might imagine from a metric that relies on people answering a single question, the NPS is both alluring and [shaky](https://www.wsj.com/articles/the-dubious-management-fad-sweeping-corporate-america-11557932084). Alluring because, amongst a sea of often confusing data points it can seemingly provide clarity. But shaky because, well, a single data point isn’t enough to go on. As a result, it’s important to combine it with other metrics, including user surveys and interviews, to really understand what’s going on.

> ### How is it useful?

Reviewer: ```suggestion
### Why is it useful?
```

> Pretty much all subscription-based businesses should be measuring their NPS as customer satisfaction is a key driver of churn. This is especially true for e-commerce businesses – if your online store is poorly designed and hard to use, your chances of repeat business will plummet. 

Reviewer: ```suggestion
Because it's an easy to understand, qualitative indicator of retention, and a simple way of measuring whether the changes you're making to your product to improve retention will be successful. If your NPS improve over time you can be reasonably confident retention should improve as well.
```

![NPS survey](https://res.cloudinary.com/dmukukwp6/image/upload/v1710055416/posthog.com/contents/images/blog/nps-csat-ces/nps-survey.png)

> This isn’t the most useful metric for very early-stage startups. NPS relies on having a large number of established customers to provide meaningful data. CSAT and CES scores, however, are very useful metrics for early-stage startups. More on those a little lower down.

Reviewer: ```suggestion
That said, this isn’t the most useful metric for very early-stage startups. NPS relies on having a large number of established customers to provide meaningful data, and scores from brand new users are less reliable and useful than those from people who have been using your product for a long time.
```

### How to calculate your NPS 

The NPS score is calculated by subtracting the percentage of Detractors from the percentage of Promoters. For example, if 80% of your customers were promoters and 5% detractors, your NPS score would be 75. Anything above 0 is good, above 20 is great, and above 50 is very good.

### What’s a good NPS?

> According to [Satmetrix](https://www.satmetrix.com/wp-content/uploads/2023/07/NICE-Net-Promoter-Benchmarks-2023.pdf), the average NPS score for software and apps is 27. But there’s a really wide range – at Dropbox, it’s 27, at Slack it’s 53, and at Square it’s 70. At PostHog, the NPS for our product analytics is 46.7, for our session replay it’s 56.6, and for feature flags it’s 65.9.

Reviewer: As a reader, I need more context here. Also, there's a fun option for a cool graphic here rather than having the answer in copy alone.

## 5. Customer satisfaction score (CSAT)

### What is a CSAT?

Your customer satisfaction score, or CSAT, is similar to your NPS, but is specifically a great customer retention metric to use before you’ve found product-market fit. Satisfaction tends to be short-lived, whereas loyalty takes time to build. And that’s why CSAT is a great metric for early-stage startups.

> ### How is it useful?

Reviewer: ```suggestion
### Why is it useful?
```

> Any businesses that interact in a very direct, tangible way with customers should be measuring their CSAT: think retailers, restaurants, and customer support. On the flipside, businesses that rely on very transactional relationships with customers (gas stations) or high-value, single-purchase businesses (car dealerships) should place higher importance on other metrics.

Reviewer: As per earlier comments, we should avoid examples about "non tech" businesses as they're not really relevant to our audience.

### How to calculate your CSAT score

Again, ask a straightforward question: ‘How simple are you with our product?’ and give people a range from 0 to 10 or 0 to 5 – or use smiley faces to represent a 0 to 5 scale. You should collect CSAT responses right after someone uses a specific feature.

![CSAT survey](https://res.cloudinary.com/dmukukwp6/image/upload/v1710055416/posthog.com/contents/images/blog/nps-csat-ces/csat-survey.png)

Here’s the math. Positive responses sit in the range of 4 to 5 if your scale is from 1 to 5 and 8, 9, or 10 if your scale is from 1 to 10. Your CSAT score is the total number of positive responses divided by the total number of responses. So if you survey 100 people and 80 people give you a positive score that’s a CSAT score of 80%.

### What’s a good CSAT score?

A good CSAT score is typically between 75% to 85%. According to [Fullview](https://www.fullview.io/blog/csat-benchmarks-by-industry#toc-csat-benchmarks-for-software-companies), the average CSAT score for SaaS companies is 78%.

## 6. Customer Effort Score (CES)

> ### What is a CES?
> 
> Another variation on the NPS and CSAT theme, your customer effort score, or CES, is a measure of how easy, or not, your product is to use. This is something of a leading indicator: a low CES score suggests you will probably have low retention because your product is hard to use – unless, miraculously, people are using it in spite of it being badly designed. As such, this is a really good metric to measure at a startup once you’ve found product-market fit.
> 
> ### How is it useful?
> 
> This is a crucial metric for e-commerce and SaaS companies. If you’ve built a product or platform that should be easy to use and spark joy, then CES is a hugely important metric. On the flipside, if your business has very limited touchpoints with customers, if you run an insurance firm, for example, then measuring your CES might not be that revelatory.

Reviewer: ```suggestion
### What is a CES?

Your customer effort score, or CES, is a measure of how easy, or not, your product is to use. This is a crucial metric if "ease of use", rather than features, is the main differentiator between your product and your competition.

### Why is it useful?

This is something of a leading indicator: a low CES score suggests you will probably have low retention because your product is hard to use – unless, miraculously, people are using it in spite of it being badly designed. As such, this is a really good metric to measure at a startup once you’ve found product-market fit.
```

  https://res.cloudinary.com/dmukukwp6/image/upload/posthog.com/contents/images/blog/posthog-marketing/marketing-hog.jpg
featuredImageType: full
tags:
  - Product analytics
  - Product metrics
  - Product engineers
  - Growth
crosspost:
  - Founders
  - Blog
---
You put a lot of work into getting people to use your product, but how much work do you put into keeping them? 

In this guide we’ll cover the key customer retention metrics, who they’re useful for, and how to calculate them. We’ve also put together some benchmarks to measure your business against, and tips on how to improve retention. 

## 1. Customer retention and churn

### What are customer retention and churn?

These are the two most important retention metrics. Customer retention is a simple measure of how well your business retains customers. Customer churn, often referred to as logo churn, is the inverse: how many customers business loses. 

Measuring monthly retention is best if you run a monthly subscriptions business. If most of your customers pay annually, measure it annually. It's useful to track both if you have a mixture of customers on monthly and annual contracts.

### Why are they useful?

> Almost any business that relies on repeat customers should be measuring retention and churn. There are some exceptions. If you run a car dealership, for example, you probably don’t need to be measuring retention and churn as cars are, typically, very rare purchases. Likewise, if you’re running a funeral home, customer retention and churn probably isn’t your most important metric.

Author: ```suggestion
Almost any business that relies on repeat customers should be measuring retention and churn. It's a simple, essential measure of company health. How churn is good, high churn is bad. Retention is the inverse and shows how sticky new customers are. Do new customers keep coming back for a week? A month? A year? And how often?Drilling down into the data will also help you see which specific features of your product are the most sticky – or hte most likely to cause churn.```

 
* **Detractors**, those who rated you between 0 and 6, could potentially damage your business through word-of-mouth.

As you might imagine from a metric that relies on people answering a single question, the NPS is both alluring and [shaky](https://www.wsj.com/articles/the-dubious-management-fad-sweeping-corporate-america-11557932084). Alluring because, amongst a sea of often confusing data points it can seemingly provide clarity. But shaky because, well, a single data point isn’t enough to go on. As a result, it’s important to combine it with other metrics, including user surveys and interviews, to really understand what’s going on.

### Why is it useful?

Because it's an easy to understand, qualitative indicator of retention, and a simple way of measuring whether the changes you're making to your product to improve retention will be successful. If your NPS improve over time you can be reasonably confident retention should improve as well.

![NPS survey](https://res.cloudinary.com/dmukukwp6/image/upload/v1710055416/posthog.com/contents/images/blog/nps-csat-ces/nps-survey.png)

> That said, this isn’t the most useful metric for very early-stage startups. NPS relies on having a large number of established customers to provide meaningful data, and scores from brand new users are less reliable and useful than those from people who have been using your product for a long time.

Reviewer: Is this what you want? Doesn't make sense now.

### How to calculate your NPS 

The NPS score is calculated by subtracting the percentage of Detractors from the percentage of Promoters. For example, if 80% of your customers were promoters and 5% detractors, your NPS score would be 75. Anything above 0 is good, above 20 is great, and above 50 is very good.

### What’s a good NPS?

> According to [Satmetrix](https://www.satmetrix.com/wp-content/uploads/2023/07/NICE-Net-Promoter-Benchmarks-2023.pdf), the average NPS score for software and apps is 27. But there’s a really wide range – at Dropbox, it’s 27, at Slack it’s 53, and at Square it’s 70. At PostHog, the NPS for our product analytics is 46.7, for our session replay it’s 56.6, and for feature flags it’s 65.9.

Author: ```suggestion
According to [Satmetrix](https://www.satmetrix.com/wp-content/uploads/2023/07/NICE-Net-Promoter-Benchmarks-2023.pdf), the average NPS score for software and apps is 27. But there’s a really wide range – at Dropbox, it’s 27, at Slack it’s 53, and at Square it’s 70. At PostHog, the NPS for our product analytics is 46.7, for our session replay it’s 56.6, and for feature flags it’s 65.9.

Below is a table with a benchmark of NPS scores from well-known SaaS products.

| Product | NPS score |
|--------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| [Snowflake](https://www.snowflake.com/blog/customer-experience-report-2022/) | 72 |
| [Square](https://d1g145x70srn7h.cloudfront.net/documents/investor-relations/presentations/05-11-2018-overview.pdf) | 70 |
| [Zoom](https://blog.zoom.us/zooms-nps-dominates-video-conferencing/) | 69 | 
| [DocuSign](https://www.docusign.com/why-docusign) | 66 |
| [Zapier](https://www.comparably.com/brands/zapier) | 64 |
| [Slack](https://www.comparably.com/brands/slack) | 53 |
| [Mailchimp](https://www.comparably.com/brands/mailchimp) | 47 |
| [Netflix](https://www.comparably.com/brands/netflix) | 46 |
| [Twilio](https://www.comparably.com/brands/twilio) | 34 |
| [Dropbox](https://www.comparably.com/brands/dropbox) | 27 | 
```

## 5. Customer satisfaction score (CSAT)

### What is a CSAT?

Your customer satisfaction score, or CSAT, is similar to your NPS, but is specifically a great customer retention metric to use before you’ve found product-market fit. Satisfaction tends to be short-lived, whereas loyalty takes time to build. And that’s why CSAT is a great metric for early-stage startups.

### Why is it useful?

> Any businesses that interact in a very direct, tangible way with customers should be measuring their CSAT: think retailers, restaurants, and customer support. On the flipside, businesses that rely on very transactional relationships with customers (gas stations) or high-value, single-purchase businesses (car dealerships) should place higher importance on other metrics.

Author: ```suggestion
Any businesses that interact in a very direct, tangible way with customers on a regular basis should be measuring their CSAT: think customer support, for example. On the flipside, businesses that rely on very transactional relationships with customers, such as online retailers, or high-value, single-purchase businesses should place higher importance on other metrics.
```

>
> Similarly, a low or high CSAT score doesn’t tell you how likely you are to retain a user. They could leave because of pricing, for example. [Dig deeper](https://newsletter.posthog.com/p/how-to-uncover-your-users-real-problems) through surveys, session replays, and user interviews to find the real pain points and the things that spark joy.
>
> Want more detail on NPS, CSAT, and CES? Check out our [in-depth guide](/product-engineers/nps-vs-csat-vs-ces).

## 7. Feature adoption rate

### What is feature adoption rate?

This is actually a whole group of metrics for all the different features of your product. Feature adoption rates measure the percentage of users who, as the name suggests, use certain features. Let’s say you’ve invested a lot of time and money developing a new product feature. Wouldn’t it suck if nobody used it? This customer retention metric helps you understand the most popular features of your product, and the ones that not many people use.

> ### How is it useful?

Author: ```suggestion
### Why is it useful?
```

> This is a super retention metric for SaaS companies, both B2C and B2B, and particularly businesses with both free and premium tiers. A good understanding of feature adoption rate will help you see which premium features are most-used, for example. On the flipside, e-commerce services, which broadly speaking have one ‘feature’ should focus on other retention metrics.

Author: ```suggestion
This is a super useful retention metric for SaaS companies, both B2C and B2B, and particularly businesses with both free and premium tiers. A good understanding of feature adoption rate will help you see which premium features are most-used, for example. On the flipside, e-commerce services, which broadly speaking have one ‘feature’, should focus on other retention metrics.
```

The formula for working this out is simple: take the number of users of a specific feature in the last month and divide it by the total number of product users, then multiply that total by 100. 

You can quickly get pretty granular here and pull out some really useful information. Combine this kind of analysis with, say, user segmentation, and you can see which segments love which features.

### What’s a good feature adoption rate?

This one is hard to benchmark as the variation between features and companies is vast. An analysis by [Pendo](https://www.pendo.io/product-benchmarks/) found that a “best in class” feature adoption rate amongst B2B SaaS companies was 28%.

## 8. DAU/MAU ratio

### What is the DAU/MAU ratio?

It’s the ratio of daily active users over monthly active users. This engagement metric measures stickiness and shows the percentage of your users who are using your product every day. If you’ve got a high DAU/MAU ratio, then people are returning to your product often. While these aren’t strictly user retention metrics, proactively addressing unexpected drops in active users or user engagement can give you more evidence of what’s working and what’s not.

> ### How is it useful?

Author: ```suggestion
### Why is it useful?
```
